src/Authentication/Authentication/Common/GraphSession.cs

```diff
@@ -21,6 +21,19 @@ public class GraphSession : IGraphSession
         /// </summary>
         public IAuthContext AuthContext { get; set; }
 
+        private byte[] token;
+
+        /// <summary>
+        /// Gets or Sets a session based token.
+        /// This returns an empty byte[] when token is not present.
+        /// </summary>
+        public byte[] Token
+        {
+            get { return token ?? new byte[0]; }
+            set { token = value; }
+        }
```

Comment:
Is this semantically equivalent to the following?
`public byte[] Token { get; set; } = new byte[0];`

Reply:
It's different, we always want the value of `Token` to be an empty byte[] when no value is present, or when value is null. `public byte[] Token { get; set; } = new byte[0];` only considers the default case, but it allows for null.